source/server/session_resource_repository.h

```diff
@@ -112,11 +113,8 @@ int SessionResourceRepository<TResourceHandle>::add_session(
   }
 
   // session_name resolves to a session in a different resource repository.
-  // Report error.
   if (!resource_map_->contains(session_from_repository)) {
-    session_id = 0;
-    const int kError_InvalidArg = 1;
-    return kError_InvalidArg;
+    throw nidevice_grpc::SessionException("The session name \"" + session_name + "\" is already being used in a different driver's service. Please use a different name.");
```

Comment:
I'm still ruminating on this message. It doesn't sound quite right because I think technically it could re-use a session from a different driver using the same resource type... I may be wrong about that though.

Also, I'm not sure if we should make a suggestion at the end of the message or not.

Reply:
Ideas:

`The session name "my_cool_sesh" is in use and refers to an incompatible driver session resource`

`The session name "hw_that_does_stuff" is being used by a different driver`

You're correct that you would not hit this for drivers with compatible session types. I.e., all IVI drivers can share sessions (at least at this level, we'll see what happens when it gets to the API) and same with RFmx. I don't think we need to be too precise about that. But yeah "in a different driver's service" is too specific/wrong and a little fluffy anyway. But I think "by a different driver" is clear enough even if it's not 100% accurate.

Reply:
Thanks, I went with the latter. It's short and accurate enough to know what's going on.